src/note.rs

```diff
@@ -193,12 +193,14 @@ impl Note {
     }
 
     /// Create a unique nullifier for the note
+    ///
+    /// This nullifier is represeted as `H(sk_r · G', pos)`
     pub fn gen_nullifier(&self, sk: &SecretSpendKey) -> BlsScalar {
         let sk_r = sk.sk_r(&self.stealth_address);
-        let sk_r = BlsScalar::from(*sk_r.as_ref());
-        let pos = BlsScalar::from(self.pos);
+        let pk_prime = GENERATOR_NUMS_EXTENDED * sk_r.as_ref();
+        let pk_prime = pk_prime.to_hash_inputs();
 
-        hash(&[sk_r, pos])
+        hash(&[pk_prime[0], pk_prime[1], BlsScalar::from(self.pos)])
```

Comment:
I would avoid to change the `pos` since it doesn't bring any difference from previous code, so let's keep:
```
let pos = BlsScalar::from(self.pos);

hash(&[pk_prime[0], pk_prime[1], pos);
```

Reply:
Fixed